### PR TITLE
Propagate labels from base image / propagate container config in case of V2.1 manifest

### LINKED ISSUE #1641: Container config not propagated when registry uses old V2.1 manifest
This normally works, but not when using `quay.io/sdase/openjdk-runtime:8-hotspot` as a base image. `docker inspect` shows that the base image has `Env`, `Labels`, etc.

This may be related to Quay using the old manifest V2.1; when I retag the base image, push it to Docker Hub (which generates manifest V2.2), and use it as a base image, it works.

### LINKED ISSUE #1643: Labels from base image are not propagated
Even if #595 marked fixed, currently labels are not being propagated. No idea if it ever worked or this is a regression. Should be an easy fix. Will fix it as part of #1641.

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #1643.
Fixes #1641.

Travis is failing to install OpenJDK 11, so ignore that for the moment.

### CODE REVIEW

File: jib-core/src/main/java/com/google/cloud/tools/jib/image/json/JsonToImageTranslator.java [hunk 1]
```diff
@@ -149,13 +161,8 @@
...
 
-    if (containerConfigurationTemplate.getContainerEntrypoint() != null) {
-      imageBuilder.setEntrypoint(containerConfigurationTemplate.getContainerEntrypoint());
-    }
-
-    if (containerConfigurationTemplate.getContainerCmd() != null) {
-      imageBuilder.setProgramArguments(containerConfigurationTemplate.getContainerCmd());
-    }
+    imageBuilder.setEntrypoint(containerConfigurationTemplate.getContainerEntrypoint());
+    imageBuilder.setProgramArguments(containerConfigurationTemplate.getContainerCmd());
```
Comment (briandealwis) [inline jib-core/src/main/java/com/google/cloud/tools/jib/image/json/JsonToImageTranslator.java]: So previously we would only overwrite entrypoint/args if set in the container configuration.  The code further above continues doing this for `created`, `architecture` and `os`, and further below.  Is there a reason to deviate from the previous pattern?

File: jib-core/src/test/resources/core/json/v21manifest.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     {"blobSum":"sha256:5bd451067f9ab05e97cda8476c82f86d9b69c2dffb60a8ad2fe3723942544ab3"}
   ],
   "history": [
-    {"v1Compatibility":"some v1-compatible object"},
+    {"v1Compatibility": "{\"architecture\":\"amd64\",\"config\":{\"Env\":[\"JAVA_HOME=/opt/openjdk\",\"PATH=/opt/openjdk/bin\"],\"Entrypoint\":[\"/opt/openjdk/bin/java\"],\"Cmd\":[\"-version\"]},\"created\":\"2019-04-17T14:02:34.79404123Z\"}"},
```
Comment (briandealwis) [inline jib-core/src/test/resources/core/json/v21manifest.json]: Would is be possible to create a new test with a this container-configuration object?  The previous test wasn't wrong: the `v1Compatibility` could be some arbitrary object.

Comment (chanseokoh) [reply jib-core/src/test/resources/core/json/v21manifest.json]: The next element is `"another v1-compatible object"`, so at least we know the method doesn't fail with an arbitrary string. I just didn't like adding the `@VisibleForTesting` method `getV1Compatibility()` used only in testing and nowhere else.

File: jib-core/src/main/java/com/google/cloud/tools/jib/image/json/JsonToImageTranslator.java (hunk 1 above)
Comment (chanseokoh) [reply jib-core/src/main/java/com/google/cloud/tools/jib/image/json/JsonToImageTranslator.java]: At some point, looks like we changed `setEntrypoint()`, `setProgramArguments()`, etc. to accept `@Nullable`. As a builder, `imageBuilder` starts with null values, and passing null to them is like no-op hence OK. It also felt a bit weird to do the a check when the method can accept null. Sure, if `setEntrypoint()` was called on `imageBuilder` with a concrete value before and if we pass null then the value will be gone, but calling `setEntrypoint()` twice with different values cannot happen here.

**Pull request merged**